sound/soc/sof/pcm.c

```diff
@@ -707,7 +707,14 @@ int sof_pcm_dai_link_fixup(struct snd_soc_pcm_runtime *rtd, struct snd_pcm_hw_pa
 		}
 		break;
 	case SOF_DAI_INTEL_ALH:
-		/* do nothing for ALH dai_link */
+		/*
```

Comment:
@RanderWang This is touching very generic part of SOF code, so the commit message needs to be a lot clearer:

- there's nothing in this patch that is specific to 2-channel or to DSM
- it seems this is just missing logic: channel acount is specified in topology, so we should read the settings here. it just happens this has not  done any harm for configs where channel count is 2
- so the 4ch configurations for SDW are example topologies that require this bugfix
- the line-wrap is somehow wrong in the commit message.. you should wrap lines at 74 columns in the decription

Unrelated note... reading this, I'm not sure why for HDA DAI we are not reading channel count here. It's specified in topology as well. But maybe a topic for another PR.

Reply:
thanks @kv2019i . There is no such issue on HDA platform and please keep eye on it.